File: algorithms/javaalgo/src/test/java/leetcode/number201to300/number241to250/number242/ValidAnagramTest.java
package leetcode.number201to300.number241to250.number242;

import static org.assertj.core.api.Assertions.assertThat;
import static org.junit.jupiter.api.Assertions.assertAll;

import java.util.Arrays;
import org.junit.jupiter.api.Test;

public class ValidAnagramTest {

    @Test
    void test() {
        String s1 = "anagram";
        String t1 = "nagaram";

        String s2 = "rat";
        String t2 = "car";

        assertAll(
            () -> assertThat(isAnagram(s1, t1)).isTrue(),
            () -> assertThat(isAnagram(s2, t2)).isFalse()
        );
    }

    public boolean isAnagram(String s, String t) {
        char[] charS = s.toCharArray();
        char[] charT = t.toCharArray();
        if (charS.length != charT.length) {
            return false;
        }

        Arrays.sort(charS);
        Arrays.sort(charT);

        for (int i = 0; i < charS.length; i++) {
            if (charS[i] != charT[i]) {
                return false;
            }
        }

        return true;
    }
}

File: algorithms/javascriptalgo/leetcode/number201-300/number241-250/number242/validAnagram.test.js
test("test", () => {
  const s1 = "anagram";
  const t1 = "nagaram";

  const s2 = "rat";
  const t2 = "car";

  expect(isAnagram(s1, t1)).toBeTruthy();
  expect(isAnagram(s2, t2)).toBeFalsy();
});

function isAnagram(s, t) {
  if(s.length !== t.length) return false;

  let sArr = s.split("").sort();
  let tArr = t.split("").sort();

  for (let i = 0; i < sArr.length; i++) {
    if (sArr[i] !== tArr[i]) {
      return false;
    }
  }

  return true;
}
